### PR TITLE
Release notes 2.11.0

### PR DESCRIPTION
**Pull request by joshmoore**
Quite a bit of text so opening the release notes for 2.11 as a PR. Feedback welcome over the next few hours. Then I'll get the release out and start work on the blogpost (https://github.com/zarr-developers/zarr-python/issues/901)

cc: @msankeys963 @d-v-b @jni @jakirkham @grlee77 @rabernat

### CODE REVIEW

**Review by jni — APPROVED** [review]

File: docs/release.rst [hunk 1]
```diff
@@ -6,28 +6,119 @@ Release notes
...
+  By :user:`Josh Moore <joshmoore>` :issue:`859`.
+
+* Drop shortcut ``fsspec[s3]`` for dependency.
+  By :user:`Josh Moore <joshmoore>` :issue:`920`.
+
+* and a swath of code-linting improvements by :user:`Dimitri Papadopoulos Orfanos <DimitriPapadopoulos>`:
+
+  - Unnecessary comprehension (:issue:`899`)
+
+  - Unnecessary ``NoneP` provided as default (:issue:`900`)
```
Comment (jakirkham) [inline docs/release.rst:101]:
```suggestion
  - Unnecessary ``None`` provided as default (:issue:`900`)
```

File: docs/release.rst [hunk 2]
```diff
@@ -6,28 +6,119 @@ Release notes
...
+
+  - use an if ``expression`` instead of `and`/`or` (:issue:`888`)
+
+  - iRemove unnecessary literal (:issue:`891`)
```
Comment (jakirkham) [inline docs/release.rst:105]:
```suggestion
  - Remove unnecessary literal (:issue:`891`)
```

File: docs/release.rst [hunk 3]
```diff
@@ -6,28 +6,119 @@ Release notes
...
+
+  - Decorate a few method with `@staticmethod` (:issue:`885`)
+
+  - Useless ``return`` (:issue:`884`)
```
Comment (jakirkham) [inline docs/release.rst:109]:
```suggestion
  - Drop unneeded ``return`` (:issue:`884`)
```

File: docs/release.rst [hunk 4]
```diff
@@ -6,28 +6,119 @@ Release notes
...
+
+  - In Python 3, all classes implicitly inherit from object (:issue:`886`)
```
Comment (jakirkham) [inline docs/release.rst:111]:
```suggestion
  - Drop explicit ``object`` inheritance from ``class``es (:issue:`886`)
```

**Review by jakirkham — COMMENTED** [review]
Thanks Josh! 😀

Few minor suggestions below

**Review by MSanKeys963 — APPROVED** [review]
Thanks a lot, @joshmoore.

Shout-out to all the contributors: @jni, @d-v-b, @grlee77, @rabernat, @Carreau, @orenwatson, @jhamman, @DimitriPapadopoulos, @raybellwaves, @jakirkham, @madsbk and @joshmoore. 🎉

Everything looks good to me.

**Pull request merged**